https://issues.redhat.com/browse/ACM-14268 (#6974)
open-cluster-management/rhacm-docs: business_continuity/backup_restore/backup_tag_resources.adoc

[#backup-tagging-resources]
= Tagging resources 

> If you intend to use an existing hub for a restore hub configuration and expect that the hub has the same content as the backup hub after the restore is completed, tag any resource that would be part of a hub backup with the `velero.io/backup-name: backupName` label annotation. Tagging the resources with the Velero annotation marks them as being generated by an earlier restore.

Reviewer: Making this suggestion to add the benefit/reason of tagging the resources
```suggestion

Tag user-created resources on the restore hub cluster to use the existing hub cluster as a restore cluster. This helps you ensure that you have the same content between the restore hub cluster and the backup hub cluster.

When you use an existing hub cluster for a restore hub cluster configuration, tag any resource that you want to be part of the hub cluster backup with the `velero.io/backup-name: backupName` label annotation. Tagging the resources with the Velero annotation marks them as being generated by an earlier restore.
```

[#prerequisite-resources]
== Prerequisite 

* Ensure you understand the outcomes for existing managed clusters by completing the documentation, xref:../backup_restore/backup_existing_clusters.adoc#backup-existing-clusters[Understanding outcomes for existing managed clusters during the restore process].
* On the restore hub, update the `DataProtectionApplication` resource and set the storage location to a new, temporary location. You do not want to create the backup used to tag the restore hub resources in the same storage location with the backup hub. 

> * Before you tag the resources, make sure you set the storage location to a different location than the backup hub.

Reviewer: Please be sure to update each instance of "hub" to "hub cluster"

[#tagging-resources-velero]
== Tagging resources with the velero label 

To tag user created resources on the restore hub with the `velero.io/backup-name: backupName` label, complete the following steps: 

. On the restore hub, create a `BackupSchedule` resource to generate a backup that has all these hub resources by using the following YAML:

+
[source,yaml]
----
apiVersion: cluster.open-cluster-management.io/v1beta1
kind: BackupSchedule
metadata:
 name: schedule-acm-msa
 namespace: open-cluster-management-backup
spec:
 veleroSchedule: 0 */1 * * *
 veleroTtl: 120h
 useManagedServiceAccount: true
 paused: false
----

> . Verify that the `BackupSchedule` has created the hub backup resources and their status is `Completed`.

Reviewer: How does the user do this? 
```suggestion
. Verify that the `BackupSchedule` created the hub cluster backup resources and the resource status is `Completed`.
```

> . Pause the `BackupSchedule` schedule and stop generating backups by setting the property `paused=true` on the `schedule-acm-msa BackupSchedule`.
> .. *Note:* You cannot run a restore operation on the hub cluster if the `BackupSchedule` is enabled, so you pause the `BackupSchedule` to run a restore operation.

Reviewer: This note is appearing as a substep and the second part of the note seems unnecessary
```suggestion
. Pause the `BackupSchedule` schedule and stop generating backups by setting the property `paused=true` on the `schedule-acm-msa BackupSchedule`.
+
*Note:* You cannot run a restore operation on the hub cluster if the `BackupSchedule` is enabled.
```

> . Run a restore operation by using the following yaml: 

Reviewer: ```suggestion
. Run a restore operation by using the following YAML: 
```

+
[source,yaml]
----
kind: Restore
metadata:
name: restore-acm
namespace: open-cluster-management-backup
spec:
cleanupBeforeRestore: None
veleroManagedClustersBackupName: latest
veleroCredentialsBackupName: latest
veleroResourcesBackupName: latest
excludedResources: # exclude managed cluster resources
 - ManagedCluster
excludedNamespaces: # exclude managed cluster namespaces
  - managed-cls-1
  - managed-cls-2
  - managed-cls-n
----

After you complete these steps, the operator goes over each resource which is part of the hub backup and tags it with the `velero.io/backup-name: backupName` , where `backupName` is the name of the latest backup. 

> After tagging the resources, you can use the hub as a restore cluster. Update the `DataProtectionApplication` storage location to point to the storage location of the backup hub so you can run a restore operation using the hub backup resources. All the restore hub resources are tagged now with the `velero.io/backup-name label`, so they show as being generated by an earlier restore. Therefore, any restore operation that you run on this hub cluster using the `cleanupBeforeRestore` option set to `CleanupRestored` cleans up the data that is not part of the restored backup.

Reviewer: Should the following statement be a part of the procedure: 

Update the `DataProtectionApplication` storage location to point to the storage location of the backup hub so you can run a restore operation using the hub backup resources.

*Important:*

> * Do not use the restore operation with the `cleanupBeforeRestore` option set to `CleanupRestored` if you have managed clusters on this hub created by an earlier restore operation which are not part of the restored backup. Otherwise, they get detached from the hub and managed cluster resources deleted. 

Reviewer: It seems like this note can be moved to the `backup_existing_clusters.adoc` file in the `#existing-clusters-detached` section

* Having this constraint, you could use an existing hub as a restore hub in the following situations:
** There are no managed clusters on this restore hub. 
** There are managed clusters on the restore hub and you do not expect the restore hub user data to be the same with the one available with the restored backup.

> ** The names of the managed clusters from the restore hub do not collide with any of the managed cluster names from the backup hub.

Reviewer: Should these be mentioned earlier as a description for Using an existing hub cluster as a restore hub cluster?
